log/logger.go

```diff
@@ -471,3 +478,203 @@ func parseSyslogAddress(location string) *syslogAddress {
 
 	return nil
 }
+
+const (
+	compressSuffix = ".gz"
+)
+
+// millRunOnce performs compression and removal of stale log files.
+// Log files are compressed if enabled via configuration and old log
+// files are removed, keeping at most l.MaxBackups files, as long as
+// none of them are older than MaxAge.
+func (l *Logger) millRunOnce() error {
+	files, err := l.oldLogFiles()
+	if err != nil {
+		return err
+	}
+
+	compress, remove := l.screeningCompressFile(files)
+
+	for _, f := range remove {
+		_ = os.Remove(filepath.Join(l.dir(), f.FileName))
+	}
+
+	for _, f := range compress {
+		var fnCompress string
+		fnCompress, err = l.findCompressFile(f.FileName)
+		if err != nil {
+			return err
+		}
+		errCompress := l.compressLogFile(f.FileName, fnCompress)
+		if err != nil && errCompress != nil {
```

Comment:
`err != nil` is useless. just `err = l.compressLogFile(f.FileName, fnCompress)` is good enough?